cmake/ChooseBlas.cmake

```diff
@@ -22,12 +22,7 @@ if(DEFINED USE_BLAS)
   set(BLAS "${USE_BLAS}")
 else()
   if(USE_MKL_IF_AVAILABLE)
-    if(NOT MKL_FOUND)
-      find_package(MKL)
-    endif()
-    if(MKL_FOUND)
-      set(BLAS "MKL")
-    endif()
+    set(BLAS "MKL")
```

Comment:
Wouldn't this fail if MKL is not available?

Reply:
It's set to 'Open' in the l.19 ('set(BLAS "Open" CACHE STRING "Selected BLAS library")'). Or I miss sth?

Reply:
Yes, it'll fail. I need to rethink it.

Reply:
Or maybe I missed something?

Reply:
Line 18 above initializes set(BLAS "Open") and only if MKL can be found, it's overwritten to BLAS=MKL.
Do you agree? If so, I think the current PR isn't needed.

With respect to USE_MKL_IF_AVAILABLE, it's fine with me to remove the variable and just check for MKL by default. As long as user doesn't specify which BLAS library they want, I think it's reasonable for MXNet to check for BLAS libraries in MXNet specified order of priority (ie. 1st prio is MKL, if not available try OpenBlas and otherwise use whatever is available..)

Reply:
> Line 18 above initializes set(BLAS "Open") and only if MKL can be found, it's overwritten to BLAS=MKL.
> Do you agree? If so, I think the current PR isn't needed.
> 
If there is no 'MKL' the following function: find_package(MKL) is looking for the MKL through of FindMKL.cmake file and produce the following warning:
```
CMake Warning at cmake/ChooseBlas.cmake:26 (find_package):
  By not providing "FindMKL.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "MKL", but
  CMake did not find one.

  Could not find a package configuration file provided by "MKL" with any of
  the following names:

    MKLConfig.cmake
    mkl-config.cmake

  Add the installation prefix of "MKL" to CMAKE_PREFIX_PATH or set "MKL_DIR"
  to a directory containing one of the above files.  If "MKL" provides a
  separate development package or SDK, be sure it has been installed.
Call Stack (most recent call first):
  CMakeLists.txt:312 (include)
```
I wanted to clear it out with that PR.

Reply:
Thanks for clarifying. One option is to move the code from `elseif(BLAS STREQUAL "MKL" OR BLAS STREQUAL "mkl")` case up here to the beginning of the file and return from file immediately if MKL is found, or proceed to the other options if MKL isn't found. Another option as you mention is to remove USE_MKL_IF_AVAILABLE. Either is fine with me